lib/encoder/src/Pololu37D.cpp

```diff
@@ -5,23 +5,52 @@
 using namespace Encoder;
 
 Pololu37D::Pololu37D(const Config &config)
-    : m_QEI(config.ChannelA, config.ChannelB, config.Index, config.quadratureEncoding),
+    : m_QEI(config.ChannelA, config.ChannelB, config.Index, GPIO::QEI::Encoding::X4_ENCODING),
       m_zeroOffsetDeg(config.offsetDeg) {}
 
 bool Pololu37D::getAngleDeg(float &angle) {
   std::scoped_lock<Mutex> lock(m_mutex);
-  angle = (m_QEI.getPulses() * m_degreesPerCount) - m_zeroOffsetDeg;
+  read();
+  angle = m_current_angle_deg;
   return true;
 }
 
 bool Pololu37D::getAngularVelocityDegPerSec(float &speed) {
   std::scoped_lock<Mutex> lock(m_mutex);
-  speed = m_QEI.getPulseVelocity_PulsesPerSec() * m_degreesPerCount;
+  read();
+  speed = m_anglular_velocity_deg_per_sec;
   return true;
 }
 
 bool Pololu37D::reset() {
   std::scoped_lock<Mutex> lock(m_mutex);
+  m_zeroOffsetDeg = m_current_angle_deg;
```

Comment:
you should do a read first to make sure `m_current_angle_deg` is updated:
```
read();
m_zeroOffsetDeg = m_current_angle_deg;
```

Reply:
done